php/WP_CLI/Iterators/CSV.php

```diff
@@ -18,7 +18,12 @@ class CSV implements \Iterator {
 	private $currentElement;
 
 	public function __construct( $filename, $delimiter = ',' ) {
-		$this->filePointer = fopen( $filename, 'r' );
+		if ( '-' === $filename && $this->has_stdin() ) {
+			$this->filePointer = fopen( 'php://stdin', 'r' );
+		} else {
+			$this->filePointer = fopen( $filename, 'r' );
+		}
```

Comment:
I think it's too magical (also known as unexpected) to interpret `-` as STDIN.

For the purposes of https://github.com/wp-cli/entity-command/issues/96, we should simply read STDIN to memory, and then parse as CSV.